CONTRIBUTING.rst

```diff
@@ -3,35 +3,91 @@ Contributing
 
+This contributing guide is trying to avoid common pitfalls, but the project
+development environment is quite common. If it's not your first rodeo, here's a TL;DR
+
+TL;DR
+-----
+
+(The following is not meant to be executed as a script)
+
+.. code-block:: console
+
+    $ # Export libpq env vars for PG connection
+    $ export PGDATABASE=procrastinate PGHOST=localhost PGUSER=postgres
+
+    $ # Launch PostgreSQL within Docker
+    $ compose up -d
+
+    $ # Explore tox entrypoints
+    $ tox -l
+
+    $ # You can do things without tox too:
+
+    $ # Install requirements
+    $ pip install -r requirements.txt
+
+    $ # Launch tests
+    $ pytest
+
+    $ # Launch demo
+    $ export PROCRASTINATE_APP=procrastinate_demo.app.app
+    $ procrastinate -h
+
 Instructions for contribution
 -----------------------------
 
+Environment variables
+^^^^^^^^^^^^^^^^^^^^^
+
+The `export` command below will be necessary whenever you want to interact with
+the database (using the project locally, launching tests, ...).
+These are standard ``libpq`` environment variables, and the values used below correspond
+to the docker setup. Feel free to adjust them as necessary.
+
+.. code-block:: console
+
+    $ export PGDATABASE=procrastinate PGHOST=localhost PGUSER=postgres
+
 Create your development database
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
+The development database can be launched using docker with a single command.
+The PostgreSQL database we used is a fresh standard out-of-the-box database
+on the latest stable version.
+
 .. code-block:: console
 
-    $ sudo apt install postgresql-client
     $ docker-compose up -d
-    $ export PGDATABASE=procrastinate PGHOST=localhost PGUSER=postgres
+
+If you want to try out the project locally, it's useful to have ``postgresql-client``
+installed. It will give you both a PostgreSQL console (``psql``) and specialized
+commands like ``createdb`` we use below.
+
+.. code-block:: console
+
+    $ sudo apt install postgresql-client
     $ createdb
 
 Set up your development environment
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-You may need to install some required packages for ``psycopg2``:
+If you don't plan to run the code interactively and just want to run tests,
+linting and build the doc, you'll just need ``tox``. You can install it
+for your user:
 
 .. code-block:: console
 
-    # Ubuntu:
-    $ apt install libpq-dev python-dev
+    $ pip install --user tox
 
-If you don't plan to run the code interactively and just want to run tests,
-linting and build the doc, you'll just need ``tox``:
+In order for this to work, you'll need to make sure your python user install binary
+directory is included in your shell's ``PATH``. One way of doing that is to add
+a line in your ``~/.profile`` (or ``~/.zprofile`` for ``zsh``). The following command
+will output the line to write in that file:
 
 .. code-block:: console
 
-    $ pip install tox
+    echo 'PATH=$(python3 -c "import site; print(site.USER_BASE)")/bin:$PATH'
```

Comment:
This can't work :D
We need the `:$PATH`at the end to not be expanded but the `$(...)` to be.
Something like:

```suggestion
    echo "PATH=$(python3 -c "import site; print(site.USER_BASE)")/bin:"'$PATH'
```